projects/plugins/jetpack/jetpack.php

```diff
@@ -135,9 +135,11 @@ function jetpack_admin_unsupported_wp_notice() { ?>
  * - If it succeeds, we require load-jetpack.php, where all legacy files are required,
  *   and where we add on to various hooks that we expect to always run.
  */
-$jetpack_autoloader = JETPACK__PLUGIN_DIR . 'vendor/autoload_packages.php';
-if ( is_readable( $jetpack_autoloader ) ) {
-	require $jetpack_autoloader;
+$jetpack_autoloader           = JETPACK__PLUGIN_DIR . 'vendor/autoload_packages.php';
+$jetpack_module_headings_file = JETPACK__PLUGIN_DIR . 'modules/module-headings.php';
+if ( is_readable( $jetpack_autoloader ) && is_readable( $jetpack_module_headings_file ) ) {
+	require_once $jetpack_autoloader;
+	require_once $jetpack_module_headings_file;
```

Comment:
My thought in changing it:
0. The file itself could be loaded earlier, so no technical reason not to.
1. Keep the check for the file and the loading of the file tightly coupled to avoid future confusion.
2. Keep the admin notice/soft fail code DRY (check/output once).

All that said, if you want it changed, no objection to changing it.

Reply:
abce129c5

Reply:
I look at it from the perspective that `load-jetpack.php` is the place where we have all the loading of these random non-autoloaded files, so it doesn't make much sense to move this one somewhere else.